src/pages/home/report/ReportActionItem.js

```diff
@@ -74,144 +73,132 @@ const propTypes = {
 
 const defaultProps = {
     draftMessage: '',
-    hasOutstandingIOU: false,
     preferredSkinTone: CONST.EMOJI_DEFAULT_SKIN_TONE,
 };
 
-class ReportActionItem extends Component {
-    constructor(props) {
-        super(props);
-        this.popoverAnchor = undefined;
-        this.state = {
-            isContextMenuActive: ReportActionContextMenu.isActiveReportAction(props.action.reportActionID),
-        };
-        this.checkIfContextMenuActive = this.checkIfContextMenuActive.bind(this);
-        this.showPopover = this.showPopover.bind(this);
-        this.renderItemContent = this.renderItemContent.bind(this);
-        this.toggleReaction = this.toggleReaction.bind(this);
-    }
-
-    shouldComponentUpdate(nextProps, nextState) {
-        return this.props.displayAsGroup !== nextProps.displayAsGroup
-            || this.props.draftMessage !== nextProps.draftMessage
-            || this.props.isMostRecentIOUReportAction !== nextProps.isMostRecentIOUReportAction
-            || this.props.hasOutstandingIOU !== nextProps.hasOutstandingIOU
-            || this.props.shouldDisplayNewMarker !== nextProps.shouldDisplayNewMarker
-            || !_.isEqual(this.props.action, nextProps.action)
-            || this.state.isContextMenuActive !== nextState.isContextMenuActive;
-    }
+function ReportActionItem(props) {
+    const [isContextMenuActive, setIsContextMenuActive] = useState(ReportActionContextMenu.isActiveReportAction(props.action.reportActionID));
+    const previousDraftMessageRef = useRef(props.draftMessage);
+    const textInputRef = useRef();
+    const popoverAnchorRef = useRef();
 
-    componentDidUpdate(prevProps) {
-        if (prevProps.draftMessage || !this.props.draftMessage) {
+    useEffect(() => {
+        if (previousDraftMessageRef.current || !props.draftMessage) {
             return;
         }
 
         // Only focus the input when user edits a message, skip it for existing drafts being edited of the report.
         // There is an animation when the comment is hidden and the edit form is shown, and there can be bugs on different mobile platforms
         // if the input is given focus in the middle of that animation which can prevent the keyboard from opening.
-        focusTextInputAfterAnimation(this.textInput, 100);
-    }
+        focusTextInputAfterAnimation(textInputRef.current, 100);
+    }, [props.draftMessage]);
 
-    checkIfContextMenuActive() {
-        this.setState({isContextMenuActive: ReportActionContextMenu.isActiveReportAction(this.props.action.reportActionID)});
-    }
+    // We need to store the previous value of draftMessage because we use it in the above useEffect hook
+    // to decide whether we should focus the text input
+    useEffect(() => {
+        previousDraftMessageRef.current = props.draftMessage;
+    }, [props.draftMessage]);
```

Comment:
Yeah it's a good question. I'm not too sure. It doesn't feel like a blocker for this PR though.

Reply:
That's actually how I had it originally 😓 but was advised against doing so. I honestly agree that we should extract it to a hook, since this use case is extremely common.